addons/info/README.md

```diff
@@ -99,14 +99,94 @@ storiesOf('Component', module)
 
 ## Usage as decorator
 
-It is possible to add infos by default to all components by using a global or story decorator. The drawback is you won't be able to display a distinct info message per story.
+It is possible to add infos by default to all components by using a global or story decorator. 
 
 It is important to declare this decorator as **the first decorator**, otherwise it won't work well.
 
 ```js
-addDecorator((story, context) => withInfo('common info')(story)(context));
+addDecorator(withInfo);
 ```
 
+Then, you can use the `info` parameter to pass certain options to your stories. 
+This can be done per book of stories:
+
+```js
+import { storiesOf } from '@storybook/react';
+
+import Component from './Component';
+
+storiesOf('Component', module)
+  .addParameters({ 
+    info: {
+      // Your settings
+    } 
+  })
+  .add('with some emoji', () => <Component/>);
+```
+
+...or for each story individually:
+```js
+import { storiesOf } from '@storybook/react';
+
+import Component from './Component';
+
+storiesOf('Component', module)
+  .add(
+    'with some emoji', 
+    () => <Component emoji/>, 
+    { info : { inline: false, header: false } } // Make your component render inline with the additional info 
+  ) 
+  .add(
+    'with no emoji', 
+    () => <Component/>, 
+    { info: '☹️ no emojis' } // Add additional info text
+  ); 
+```
+
+...or even together:
+
+```js
+import { storiesOf } from '@storybook/react';
+
+import Component from './Component';
+
+storiesOf('Component', module)
+  .addParameters({ 
+    info: {               // Make a default for all stories in this book,
+      inline: true,       // where the components are inlined
+      styles: {
+        header: {
+          h1: {
+            color: 'red'  // and the headers of the sections are red.
+          }
+        }
+      },
+    } 
+  })
+  .add(
+    'green version', 
+    () => <Component green/>,
+    { 
+      info: { 
+        styles: { 
+          headers: { 
+            h1: { 
+              color: 'green' // Still inlined but with green headers!
+            } 
+          } 
+        } 
+      } 
+    })
+  .add(
+    'something else',
+    () => <Component different/>,
+    {
+      info: "This story has additional text added to the info!" // Still inlined and with red headers!
+    }
+  );
+```
+
+
 ## Global options
```

Comment:
Let's deprecate this and the usage as a HoC ("Basic usage" and "Usage with options") -- so let's remove it from the docs.

Reply:
@tmeasday on a related note, do you think we should have a `Deprecated` section in `storybook-official` so we still get story snapshots for deprecated usage, but they are separate from the recommended usage examples?

Reply:
@tmeasday Just so we're on the same line: you want to deprecate the `setDefaults` way of setting defaults (and shift that to the `addDecorator` in the `config.js` with either options or parameters) and remove all the docs involving this + the HoC?

Related to this, should I then rewrite all the existing `info` stories in `official-storybook` to use the parameterized way?

Reply:
Sorry guys, missed this!

> should have a Deprecated section in storybook-official so we still get story snapshots for deprecated usage, but they are separate from the recommended usage examples?

Probably a good idea @shilman -- perhaps a small refactor of the stories after we get the last few PRs merged is in order.

> Just so we're on the same line: you want to deprecate the setDefaults way of setting defaults (and shift that to the addDecorator in the config.js with either options or parameters) and remove all the docs involving this + the HoC?

Yeah that was what I was thinking, do you agree? Definitely get rid of the HoC; and `setDefaults` seems redundant now, right?

> Related to this, should I then rewrite all the existing info stories in official-storybook to use the parameterized way?

What I've been doing is leaving the existing ones as "tests" for the deprecated functionality (we can remove them when actually delete that functionality. Then making sure there are stories to cover all the new ways of doing it! You don't need to re-test that parameter inheritance works though (although maybe adding a the story in `core.stories.js` to test your change to the way it works?), just add stories that make sense.

I hope I'm making sense here 😬